app/uk/gov/hmrc/agentmapping/connector/EnrolmentStoreProxyConnector.scala

```diff
@@ -41,34 +41,37 @@ case class EnrolmentResponse(enrolments: Seq[Enrolment])
 class EnrolmentStoreProxyConnector @Inject()(appConfig: AppConfig, http: HttpClient, metrics: Metrics)(
   implicit ec: ExecutionContext) {
 
-  private val batchSize = appConfig.clientCountMaxResults
+  private val batchSize = appConfig.clientCountBatchSize
+  private val maxClientRelationships = appConfig.clientCountMaxResults
 
   private val espBaseUrl = appConfig.enrolmentStoreProxyBaseUrl
 
-  def getClientCount(userId: String)(implicit hc: HeaderCarrier): Future[Int] = {
-
-    val vatCountF = getClientCount(userId, "HMCE-VATDEC-ORG")
-    val irsaCountF = getClientCount(userId, "IR-SA")
+  private val HMCE_VATDEC_ORG = "HMCE-VATDEC-ORG"
+  private val IR_SA = "IR-SA"
 
+  def getClientCount(userId: String)(implicit hc: HeaderCarrier): Future[Int] =
     for {
-      vatCount  <- vatCountF
-      irService <- irsaCountF
-    } yield vatCount + irService
-  }
+      vatF  <- getClientCount(userId, HMCE_VATDEC_ORG)
+      irSaF <- getClientCount(userId, IR_SA, vatF)
+    } yield irSaF
 
-  private def getClientCount(userId: String, service: String)(implicit hc: HeaderCarrier): Future[Int] = {
+  private def getClientCount(userId: String, service: String, cumulativeCount: Int = 0)(
+    implicit hc: HeaderCarrier): Future[Int] = {
 
     @SuppressWarnings(Array("org.wartremover.warts.Recursion"))
-    def doGet(cumCount: Int = 0, startRecord: Int = 1): Future[Int] =
-      getDelegatedEnrolmentsCountFor(userId, startRecord, service)
-        .flatMap {
-          case (preFilteredCount, batchCount) =>
-            if (preFilteredCount < batchSize) {
+    def doGet(cumCount: Int = cumulativeCount, startRecord: Int = 1): Future[Int] = cumCount match {
+      case c if c >= maxClientRelationships => maxClientRelationships
+      case _ =>
+        getDelegatedEnrolmentsCountFor(userId, startRecord, service).flatMap {
+          case (prefilterteredCount, batchCount) =>
+            if (prefilterteredCount < batchSize) {
               batchCount + cumCount
             } else
               doGet(batchCount + cumCount, startRecord + batchCount)
```

Comment:
Can we check this is right?  why don't we move the start pointer along by batchSize, rather than batchCount?  What if the filtered count is less than the batchCount... I think the start pointer won't move along right... thoughts?

Reply:
Good spot! I believe you're correct we would be double counting some records unless we progress along with the batchSize. I've changed this and updated the tests to include some records that are not Active.